### PR TITLE
Add `foot_take_off_acceleration` and `foot_take_off_velocity` parameters in the `SwingFootPlanner` class

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This PR introduces `foot_take_off_acceleration` and `foot_take_off_velocity` parameters in the `SwingFootPlanner` class.

I also remove `std::cerr` in favour if `log()->error(...)`

### CODE REVIEW

File: src/Planners/src/SwingFootPlanner.cpp [hunk 1]
```diff
@@ -164,9 +178,11 @@ bool SwingFootPlanner::updateSE3Traj()
 
 bool SwingFootPlanner::advance()
 {
+    constexpr auto logPrefix = "[SwingFootPlanner::advance]";
+
     // update the time taking into account the limits
-    m_currentTrajectoryTime
-        = std::min(m_dT + m_currentTrajectoryTime, m_contactList.lastContact()->deactivationTime);
+    m_currentTrajectoryTime = std::min(m_dT + m_currentTrajectoryTime, //
```
Comment (S-Dafarra) [inline src/Planners/src/SwingFootPlanner.cpp:184]:
```suggestion
    m_currentTrajectoryTime = std::min(m_dT + m_currentTrajectoryTime,
```

File: src/Planners/src/SwingFootPlanner.cpp [hunk 2]
```diff
@@ -212,16 +228,17 @@ bool SwingFootPlanner::advance()
...
-            std::cerr << "[SwingFootPlanner::advance] Unable to set the knots for the planar "
-                         "planner."
-                      << std::endl;
+            log()->error("{} Unable to set the knots for the planar planner.", logPrefix);
             return false;
         }
 
-        const double footHeightViaPointPos = (m_currentContactPtr->pose.translation()(2) +
-                                              nextContactPtr->pose.translation()(2)) / 2.0 + m_stepHeight;
+        const double footHeightViaPointPos = (m_currentContactPtr->pose.translation()(2) //
```
Comment (S-Dafarra) [inline src/Planners/src/SwingFootPlanner.cpp:235]:
```suggestion
        const double footHeightViaPointPos = (m_currentContactPtr->pose.translation()(2)
```

File: src/Planners/src/SwingFootPlanner.cpp [hunk 3]
```diff
@@ -212,16 +228,17 @@ bool SwingFootPlanner::advance()
...
+                                              + nextContactPtr->pose.translation()(2))
+                                                 / 2.0
+                                             + m_stepHeight;
 
-        const double footHeightViaPointTime = m_footApexTime * T + m_currentContactPtr->deactivationTime;
+        const double footHeightViaPointTime = m_footApexTime * T //
```
Comment (S-Dafarra) [inline src/Planners/src/SwingFootPlanner.cpp:240]:
```suggestion
        const double footHeightViaPointTime = m_footApexTime * T
```

**Review by S-Dafarra — APPROVED** [review]
I noticed that you are adding some ``//`` before ending a line. Not sure if it is wanted

**Comment by GiulioRomualdi** [comment]
Hi @S-Dafarra `// ` are useful to trick clang format to have more readable code

**Comment by S-Dafarra** [comment]
> Hi @S-Dafarra `// ` are useful to trick clang format to have more readable code
> 
> I discovered it from @diegoferigo

Ok, then ignore the comments above

**Review by paolo-viceconte — APPROVED** [review]

**Comment by diegoferigo** [comment]
> Hi @S-Dafarra `// ` are useful to trick clang format to have more readable code
> 
> I discovered it from @diegoferigo

clang-format uses a cost-based scheme in edge cases to select the less worst case between options. For how blf is configured, a long function names with string arguments gets broken weirdly, and I mean that the string is split in two parts instead of going all together to a new line. This is quite odd and I do not exclude that the configuration has either something off or something that can be optimized. This is not the case of the current PR, tho it could be related.

**Comment by diegoferigo** [comment]
I didn't review this PR, but next time could we try to make these changes backward compatible e.g. by using default values? I just realized that all the code using the `SwingFootPlanner` has a partial configuration now.

**Pull request merged**